api/helpers/errors.js

```diff
@@ -44,10 +46,87 @@ class ILSIntegrationError extends Error {
   }
 }
 
+class PatronNotFound extends Error {
+  constructor() {
+    super();
+    this.type = "patron-not-found";
+    this.name = "PatronNotFound";
+    this.message = "The patron couldn't be found.";
+    this.status = 502;
+  }
+}
+
+class NoBarcode extends Error {
+  constructor(message) {
+    super();
+    this.type = "no-barcode";
+    this.name = "NoBarcode";
+    this.message = message;
+    this.status = 502;
+  }
+}
+
+class DatabaseError extends Error {
+  constructor(message) {
+    super();
+    this.type = "database-error";
+    this.name = "DatabaseError";
+    this.message = message;
+    this.status = 500;
+  }
+}
+
+class MissingRequiredValues extends Error {
+  constructor(message) {
+    super();
+    this.type = "missing-required-values";
+    this.name = "MissingRequiredValues";
+    this.message = message;
+    this.status = 400;
+  }
+}
+class IncorrectPin extends Error {
+  constructor() {
+    super();
+    this.type = "incorrect-pin";
+    this.name = "MissingRequiredValues";
+    this.message =
+      "PIN should be 4 numeric characters only. Please revise your PIN.";
+    this.status = 400;
+  }
+}
+
+class ExpiredAccount extends Error {
+  constructor() {
+    super();
+    this.type = "expired-account";
+    this.name = "ExpiredAccount";
+    this.message = "Your card has expired. Please try applying again.";
```

Comment:
This phrasing seems potentially confusing--won't people think it means they should just retry the thing they just did that led to this error?

Reply:
That's true. @killerpuppytails , I got this from the document. Should it be updated or should the clients handle this error?

Reply:
This is actually the same language that would be used by Ask. Therefore, no update. If we need to, the clients can include custom error text.

Reply:
In that case, I think this is all set!